File: ruby/src/Creational/Factory_Method/HonoluluSoupFactoryMethodSubclass.rb


#//HonoluluSoupFactoryMethodSubclass - Two of Two Subclass Factory Methods
require "Creational/Factory_Method/SoupFactoryMethod"
require "common/HonoluluClamChowder"
require "common/HonoluluFishChowder"

class HonoluluSoupFactoryMethodSubclass < SoupFactoryMethod
	def makeBuffetName()
		return "Honolulu Soup Buffet"
	end

	def makeClamChowder()
  		return HonoluluClamChowder.new()
	end

	def makeFishChowder()
		return HonoluluFishChowder.new()
	end

end

File: csharp/src/Creational/Factory_Method/HonoluluSoupFactoryMethodSubclass.cs
//HonoluluSoupFactoryMethodSubclass.java - Two of Two Subclass Factory Methods


namespace Creational.Factory_Method {

using common;

public class HonoluluSoupFactoryMethodSubclass : SoupFactoryMethod {
    public new string makeBuffetName() {
        return "Honolulu Soup Buffet";
    }
    public new ClamChowder makeClamChowder() {
        return new HonoluluClamChowder();
    }
    public new FishChowder makeFishChowder() {
        return new HonoluluFishChowder();
    }
}

}
